File: cpp/563.cpp
#include <cstdlib>
#include <iostream>

using namespace std;

// Definition for a binary tree node.
struct TreeNode {
  int val;
  TreeNode *left;
  TreeNode *right;
  TreeNode() : val(0), left(nullptr), right(nullptr) {}
  TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
  TreeNode(int x, TreeNode *left, TreeNode *right)
      : val(x), left(left), right(right) {}
};

class Solution {
public:
  int ret;
  int helper(TreeNode *cur) {
    if (!cur)
      return 0;
    int ls = helper(cur->left), rs = helper(cur->right);
    ret += abs(ls - rs);
    return cur->val + ls + rs;
  }
  int findTilt(TreeNode *root) {
    ret = 0;
    helper(root);
    return ret;
  }
};

File: go/563.go
package main

import "math"

// Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func helper(t *TreeNode) int {
	if t == nil {
		return 0
	}
	ls, rs := helper(t.Left), helper(t.Right)
	ret += int(math.Abs(float64(ls - rs)))
	return t.Val + ls + rs
}

var ret int

func findTilt(root *TreeNode) int {
	ret = 0
	helper(root)
	return ret
}
